lib/matplotlib/axes/_axes.py

```diff
@@ -6142,6 +6147,16 @@ def _normalize_input(inp, ename='input'):
                 raise ValueError(
                     'weights should have the same shape as x')
 
+        # Combine the masks from x[i] and w[i] (if applicable) into a single
+        # mask and apply it to both.
+        if not input_empty:
+            for i in range(len(x)):
+                mask_i = x[i].mask
+                if w[i] is not None:
+                    mask_i = mask_i | w[i].mask
```

Comment:
I agree; combining independent masks on both via a union is not difficult, it is easy to explain, and it handles every situation that might arise.
See `cbook.delete_masked_points`; you could use this as soon as you have the list of sequences of values and the matching list of sequences of weights or of Nones.